cmake/toolchains/c.windows.common.cmake

```diff
@@ -26,4 +26,7 @@ set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 
+list(APPEND CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES ${CMAKE_FIND_ROOT_PATH}/include)
```

Comment:
CMAKE_SYSROOT seems to be the better solution: https://gitlab.kitware.com/cmake/cmake/issues/16291#note_440550

Reply:
nope, that caused to set --sysroot on the compiler, removing /usr/lib/gcc/xxxx/... completly

Reply:
oh and it does not work:

```
d /workdir/build/external/libutil/nowide/src && /usr/bin/ccache /usr/lib/ccache/i686-w64-mingw32-g++ --sysroot=/usr/i686-w64-mingw32  -DBOOST_NOWIDE_NO_LIB -DBOOST_NOWIDE_USE_FILESYSTEM -DBOOST_PP_VARIADICS=1 -DNOMINMAX -DUNICODE -D_UNICODE -D__USE_W32_SOCKETS -isystem /workdir/external/libutil/nowide/include -isystem /usr/i686-w64-mingw32/include  -mfpmath=sse -msse -O2 -g -DNDEBUG -ggdb   -o CMakeFiles/nowide-static.dir/iostream.cpp.obj -c /workdir/external/libutil/nowide/src/iostream.cpp
In file included from /usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/ext/string_conversions.h:41:0,
                 from /usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/bits/basic_string.h:6349,
                 from /usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/string:52,
                 from /usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/stdexcept:39,
                 from /usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/array:39,
                 from /usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/tuple:39,
                 from /usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/bits/unique_ptr.h:37,
                 from /usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/memory:80,
                 from /usr/i686-w64-mingw32/include/boost/smart_ptr/scoped_ptr.hpp:22,
                 from /usr/i686-w64-mingw32/include/boost/scoped_ptr.hpp:13,
                 from /workdir/external/libutil/nowide/include/boost/nowide/iostream.hpp:12,
                 from /workdir/external/libutil/nowide/src/iostream.cpp:9:
/usr/lib/gcc/i686-w64-mingw32/7.3-win32/include/c++/cstdlib:75:15: fatal error: stdlib.h: No such file or directory
 #include_next <stdlib.h>

```

Reply:
the problem is manually adding isystem paths, include_next will jump to the wrong include path then if -isystem /usr.../include is in there (would happen on linux too if cmake would add /usr/include for example)

Reply:
/usr/include is host system. /usr/$platform/include is correct, but cmake does not have them in the implicit rules automatically if you set a CMAKE_FIND_ROOT_PATH

Reply:
Sorry I meant in the toolchain.
Why toolchain/include and not toolchain/usr/include?